### PR TITLE
AMP creative inabox Server Side Rendering image inclusion

### PR DESCRIPTION
**Pull request by keithwrightbos**
As part of Server Side Rendering for inabox AMP creatives, the creative will be transformed to include the image directly within amp-image/amp-pixel tags to ensure immediate fetch/render by the browser.  This PR ensures that the existence of the img tag prevents duplicate image fetch.

### CODE REVIEW

File: builtins/amp-img.js [hunk 1]
```diff
@@ -122,7 +122,12 @@ export class AmpImg extends BaseElement {
       this.allowImgLoadFallback_ = false;
     }
 
-    this.img_ = new Image();
+    // For inabox SSR, image will have been written directly to DOM so no need
+    // to recreate.  Calling appendChild again will have no effect.
+    if (this.element.hasAttribute('i-amphtml-ssr')) {
+      this.img_ = this.element.querySelector('img');
```
Comment (dvoytenko) [inline builtins/amp-img.js]: Use `scopedQuerySelector` from dom.js - it's pretty important.

File: builtins/amp-pixel.js [hunk 2]
```diff
@@ -59,6 +59,11 @@ export class AmpPixel extends BaseElement {
           `${TAG}: invalid "referrerpolicy" value "${this.referrerPolicy_}".`
           + ' Only "no-referrer" is supported');
     }
+    if (this.element.hasAttribute('i-amphtml-ssr') &&
+        this.element.querySelector('img')) {
```
Comment (dvoytenko) [inline builtins/amp-pixel.js]: scopedQuerySelector

File: builtins/amp-img.js (hunk 1 above)
Comment (keithwrightbos) [reply builtins/amp-img.js]: Done

File: builtins/amp-pixel.js (hunk 2 above)
Comment (keithwrightbos) [reply builtins/amp-pixel.js]: Done

**Comment by keithwrightbos** [comment]
@dvoytenko PTAL.  thanks

**Pull request merged**